src/client/unittests/common/managers/baseTestManager.ts

```diff
@@ -46,7 +47,7 @@ export abstract class BaseTestManager implements ITestManager {
     }
     constructor(public readonly testProvider: TestProvider, private product: Product, public readonly workspaceFolder: Uri, protected rootDirectory: string,
         protected serviceContainer: IServiceContainer) {
-        this._status = TestStatus.Unknown;
+        this.updateStatus(TestStatus.Unknown);
```

Comment:
There will be no listeners at this point, shouldn't we just set the internal status? With the timeout that is incurred and the event being raised perhaps it's best to not go through the `updateStatus` method *at construction time*.

```suggestion
        this._status = TestStatus.Unknown;
```

Reply:
Absolutely no harm with the current approach, consistent way of updating the status, else we have one way in one place and another in other place.